research/spoofax-org-export.tex

```diff
@@ -0,0 +1,234 @@
+Spoofax is a platform that allows for the completely \emph{declarative}
+definition of a programming language and accompanying IDE
+support~\cite{Kats10a}. Such a platform is called a \emph{language
+workbench}. The definition of a programming language is done using
+high-level \emph{meta-languages} for each aspect of the programming
+language.
+
+To define a language declaratively means that one uses the
+meta-languages to specify \emph{what} the properties of a language are, and
+not \emph{how} these properties are implemented. For example, instead of
+asking "How do I implement a tokenizer and parser for my language?",
+one asks "What is the syntax of my language?". From such a description
+in a meta-language, the tokenizer and parser can be derived, without
+the designer of the language ever having to care about its
+implementation.
+
+This section goes over the parts that make up the specification of a
+language, and for each part the relevant part of Spoofax is
+given\footnote{This section follows the structure of the
+language specification portion of the compiler construction course at
+the TU Delft. The slides can be found here:
+\url{http://tudelft-in4303.github.io/lectures/specification/}.}. The parts
+of a language specification are:
+\begin{enumerate}
+\item \hyperref[sec-syntax-def]{Syntax Definition}: Specifying the syntax of a language.
+\item \hyperref[sec-static-analysis]{Static Semantics}: Describing the static analysis part of a
+language: type checking, name binding and variable scoping.
+\item \hyperref[sec-term-rewrite]{Term Rewriting and Program Transformation}: Rewriting ASTs to new
+ASTs, for example to declare desugaring rules.
+\item \hyperref[sec-dynamic-semantics]{Dynamic Semantics}: Defining what the language does upon execution.
+\end{enumerate}
+
+\subsection{Syntax Definition}
+\label{sec-syntax-def}
+The first part of the specification of a language is its syntax. The
+syntax of a language is often specified by means of a \emph{lexical
+grammar} and a \emph{context-free grammar}, as can be seen in the
+specification of, for example, Standard ML~\cite{Milner97}. The
+lexical grammar is most often defined using regular expression. It
+defines the individual words made up of characters, such as
```

Comment:
noone knows what that is lol. The fact that you have to link it says something :P